sherpa/models/regrid.py

```diff
@@ -538,8 +448,27 @@ def __init__(self, evaluation_space=None, name='regrid1d', **kwargs):
 
         self.method = kwargs.get("interp", akima)
 
+    @property
+    def method(self):
+        """Interpolate the data from the internal to requested grid.
+
+        This is *only* used for point grids, as integrated grids use a
+        simple rebinning scheme. The default is
+        sherpa.utils.akima.akima.  The callable should accept `(xout,
+        xin, yin)` arguments and interpolate the `(xin, yin)` data
+        onto the `xout` grid, returning the interpolated data.
+        """
+        return self._method
+
+    @method.setter
+    def method(self, method):
+        if not callable(method):
+            raise TypeError(f"method argument '{repr(method)}' is not callable")
```

Comment:
Fine. But of course Python would fail later anyway when it tries to call that argument and it's not callable.

Reply:
I prefer to catch an error like this earlier - close to the error - rather than later when the "this is not callable" error gets raised by some code unrelated to this setting.

Reply:
As I said, it's "fine". I just wanted to point out that it's not strictly necessary.